proof/refine/ARM/Syscall_R.thy

```diff
@@ -1522,72 +1522,181 @@ crunch sch_act_sane[wp]: cteDeleteOne sch_act_sane
-          apply (simp add: lookup_cap_def lookup_slot_for_thread_def)
-          apply wp
-           apply (simp add: split_def)
-           apply (wp resolve_address_bits_valid_fault2)+
-         apply (wp getNotification_wp | wpcw | simp add: valid_fault_def whenE_def split del: if_split)+
-  apply (clarsimp simp add: ct_in_state_def  ct_in_state'_def conj_comms invs_valid_tcb_ctable
-                            invs_valid_objs tcb_at_invs invs_psp_aligned invs_cur)
-  apply (clarsimp simp: invs'_def valid_state'_def valid_pspace'_def
-                        ct_in_state'_def sch_act_sane_not)
-  done *)
+               apply (rule receiveIPC_corres)
+                  apply ((clarsimp simp: cap_relation_def ct_in_state_def)+)[6]
+            apply (rename_tac rights)
+            apply (simp add: bool.case_eq_if if_swap[where P="AllowRead \<in> x" for x, symmetric] split del: if_split)
+            apply (case_tac "AllowRead \<in> rights")
+             apply clarsimp
+             apply (rule stronger_corres_guard_imp)
+               apply (rule corres_split_liftEE[OF getBoundBNotification_corres])
+                 apply (case_tac "rv = Some thread \<or> rv = None")
+                  apply simp
+                  apply (rule receiveSignal_corres)
+                   apply simp
+                  apply (clarsimp simp: cap_relation_def)
+                 apply (clarsimp simp: lookup_failure_map_def)
+                apply (wpsimp wp: get_sk_obj_ref_wp getNotification_wp)+
+              apply (clarsimp simp: valid_cap_def valid_sched_def valid_sched_action_def
+                                    current_time_bounded_def ct_in_state_def)
+             apply (clarsimp simp: valid_cap_def valid_cap'_def dest!: state_relationD)
+            apply (clarsimp simp: lookup_failure_map_def)
+           apply (wpsimp wp: get_sk_obj_ref_wp)+
+         apply (rule_tac Q="\<lambda>_. ?pre1 and (\<lambda>s. cur_thread s = thread)
+                                 and K (valid_fault (ExceptionTypes_A.fault.CapFault x True
+                                        (ExceptionTypes_A.lookup_failure.MissingCapability 0)))"
+                     and E=E and F=E for E
+                in hoare_post_impErr[rotated])
+           apply (fastforce simp: valid_sched_valid_sched_action valid_sched_active_sc_valid_refills ct_in_state_def)
+          apply simp
+         apply (wpsimp wp: resolve_address_bits_valid_fault2 simp: lookup_cap_def lookup_cap_and_slot_def lookup_slot_for_thread_def)
+        apply wp
+         apply (case_tac epCap; simp split del: if_split)
+                      apply wpsimp
+                     apply wpsimp
+                    apply (rename_tac readright; case_tac readright; simp)
+                     apply wp
+                       apply simp
+                       apply wp
+                      apply (wp getNotification_wp)
+                     apply clarsimp
+                    apply (wpsimp+)[13]
```

Comment:
Is there no way to avoid this [13]?

Reply:
It looks like you managed to avoid it, nice!